tests/test_alter.py

```diff
@@ -39,3 +39,26 @@ def test_alter():
     restored = schema.connection.query("SHOW CREATE TABLE " + Experiment.full_table_name).fetchone()[1]
     assert_not_equal(altered, restored)
     assert_equal(original, restored)
+
+
+@schema
+class AlterMaster(dj.Manual):
+    definition = """
+    master_id : int
+    """
+
+    class AlterPart(dj.Part):
+        definition = """
+        -> master
+        """
+
+        definition1 = """
+        -> AlterMaster
+        """
+
+def test_alter_part_master_keyword():
+    original = schema.connection.query("SHOW CREATE TABLE " + AlterMaster.AlterPart.full_table_name).fetchone()[1]
+    AlterMaster.AlterPart.definition = AlterMaster.AlterPart.definition1
+    AlterMaster.AlterPart.alter(prompt=False)
+    altered = schema.connection.query("SHOW CREATE TABLE " + AlterMaster.AlterPart.full_table_name).fetchone()[1]
+    assert_equal(original, altered)
```

Comment:
A good principle to guide you through fixes and new features is [test-driven development](https://en.wikipedia.org/wiki/Test-driven_development). We use this quite a bit throughout the DataJoint codebases.

Here I would say that your test violates this since it still passes even before a fix is introduced. We should think first of a breaking test case based on the reference issue and work backwards from there to find the solution.

Also, we typically keep the table declarations together near the top as opposed to weaving them between tests. Taking this into account and your [comment](https://github.com/datajoint/datajoint-python/pull/991#issuecomment-1033694920) of another edge-case, we can create one like so:

```suggestion
@schema
class Parent(dj.Manual):
    definition = """
    parent_id: int
    """

    class Child(dj.Part):
        definition = """
        -> Parent
        """
        definition_new = """
        -> master
        ---
        child_id=null: int
        """

    class Grandchild(dj.Part):
        definition = """
        -> master.Child
        """
        definition_new = """
        -> master.Child
        ---
        grandchild_id=null: int
        """
# --- other tests ---
def test_alter_part():
    # https://github.com/datajoint/datajoint-python/issues/936
    def verify_alter(table, attribute_sql):
        definition_original = schema.connection.query(
            f"SHOW CREATE TABLE {table.full_table_name}"
        ).fetchone()[1]
        table.definition = table.definition_new
        table.alter(prompt=False)
        definition_new = schema.connection.query(
            f"SHOW CREATE TABLE {table.full_table_name}"
        ).fetchone()[1]
        assert (
            definition_new.replace(f"{attribute_sql},\n  ", "") == definition_original
        )

    verify_alter(table=Parent.Child, attribute_sql="`child_id` int DEFAULT NULL")
    verify_alter(
        table=Parent.Grandchild, attribute_sql="`grandchild_id` int DEFAULT NULL"
    )
```

Reply:
Great, thanks! I made this change.